src/Object/Contact.php

```diff
@@ -619,9 +594,10 @@ public static function getIdForURL($url, $uid = 0, $no_update = false)
 				'confirm' => $data["confirm"], 'poco' => $data["poco"],
 				'name-date' => datetime_convert(), 'uri-date' => datetime_convert(),
 				'avatar-date' => datetime_convert(), 'writable' => 1, 'blocked' => 0,
-				'readonly' => 0, 'pending' => 0));
+				'readonly' => 0, 'pending' => 0)
+			);
 
-			$contacts = q("SELECT `id` FROM `contact` WHERE `nurl` = '%s' AND `uid` = %d ORDER BY `id` LIMIT 2", dbesc(normalise_link($data["url"])), intval($uid));
+			$contacts = dba::select('contact', array('id'), array('nurl' => normalise_link($data["url"]), 'uid' => $uid), array('order' => 'id', 'limit' => 2));
```

Comment:
The ```dba::select``` doesn't return an array like the ```q``` function but an object that has to be fetched via ```dba::fetch```, for example:

```
while ($row = dba::fetch($r)) {
    ....
}
```

Alternatively you could put the ```$contacts``` variable through the method ```dba::inArray``` which then generates an array in the format that the ```q``` function generates.